From da22d9d27642370278a219531f2c9e37b184ce57 Mon Sep 17 00:00:00 2001
From: Bowgo Tsai <bowgotsai@google.com>
Date: Fri, 29 Jun 2018 10:15:34 +0800
Subject: [PATCH 28/31] Add /mnt/product rw mount point for product partitions.

1. init creates /mnt/product used to mount product-specific rw partitions.
2. If a device tree fstab entry for early mount specifies a mount point
   under /mnt/product e.g. /mnt/product/foo, init will create
   /mnt/product/foo mount point.

Bug: 110808288
Test: change dt fstab entry to mount /mnt/product/foo; mount point is
      created correctly, and partition is mounted in early mount.

Change-Id: I321e314992abe1084fd67a382c205f5c0c92bf3d
Merged-In: I321e314992abe1084fd67a382c205f5c0c92bf3d
(cherry picked from commit acc31cee1bd5fe270a22f993fdff79d83f827658)
---
 init/init.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/init/init.cpp b/init/init.cpp
index e57c6c6ffc..53eaa61cbc 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -631,6 +631,9 @@ int main(int argc, char** argv) {
         // /mnt/vendor is used to mount vendor-specific partitions that can not be
         // part of the vendor partition, e.g. because they are mounted read-write.
         mkdir("/mnt/vendor", 0755);
+        // /mnt/product is used to mount product-specific partitions that can not be
+        // part of the product partition, e.g. because they are mounted read-write.
+        mkdir("/mnt/product", 0755);
 
         // Now that tmpfs is mounted on /dev and we have /dev/kmsg, we can actually
         // talk to the outside world...
-- 
2.17.1

